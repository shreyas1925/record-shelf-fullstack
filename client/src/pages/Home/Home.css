.home--container{
    height:90vh;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
}

.home--container p{
    font-size:22px;
    margin-top: 10px;
    font-weight: lighter;
}

.home--container button{
    margin-top: 20px;
    padding: 10px 20px;
    border: none;
    border-radius: 5px;
    background-color: #4C63AA;
    color: white;
    font-weight: 800;
    font-size: 16px;
    cursor: pointer;
    width:120px;
    border-radius: 20px;
}

.home--container button > a{
    text-decoration: none;
    color: white;
}