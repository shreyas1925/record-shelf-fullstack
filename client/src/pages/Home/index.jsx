import React from 'react'
import "./Home.css"
import { Link } from 'react-router-dom'
const Home = () => {
  return (
    <div className='home--container' data-testid='home'>
        <p>:((</p>
        <p>seems a bit empty in here...</p>
        <button>
            <Link to="/songs">sync</Link>
        </button>
    </div>
  )
}

export default Home