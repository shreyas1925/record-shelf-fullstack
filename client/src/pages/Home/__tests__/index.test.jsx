
import {  render, screen } from '@testing-library/react';
import Home from '..';
import { BrowserRouter } from 'react-router-dom';

describe('Home Snapshot', () => {
    it('should render forming a snapshot', () => {
        const { asFragment } = render(<BrowserRouter><Home/></BrowserRouter>);
        expect(asFragment()).toMatchSnapshot(); 
    });
});
describe('Home', () => {
    it('should render without crashing', () => {
        render(<BrowserRouter><Home/></BrowserRouter>);
        const homeElement = screen.getByTestId('home');
        expect(homeElement).toBeInTheDocument();
    
    });
    it('should display data after data rendering', () => {
        render(<BrowserRouter><Home/></BrowserRouter>);
        const homeElement = screen.getByText('seems a bit empty in here...',{exact:false});
        expect(homeElement).toBeInTheDocument();
    });
    it('should navigate to songs page after clicking \'sync\'', () => {
        render(<BrowserRouter><Home/></BrowserRouter>);
        const linkElement = screen.getByRole('link', { name: 'sync' });
        expect(linkElement).toHaveAttribute('href', '/songs');
    });
});