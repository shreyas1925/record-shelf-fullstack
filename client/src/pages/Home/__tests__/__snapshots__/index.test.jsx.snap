// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Snapshot should render forming a snapshot 1`] = `
<DocumentFragment>
  <div
    class="home--container"
    data-testid="home"
  >
    <p>
      :((
    </p>
    <p>
      seems a bit empty in here...
    </p>
    <button>
      <a
        href="/songs"
      >
        sync
      </a>
    </button>
  </div>
</DocumentFragment>
`;
